zip-0312.rst

```diff
@@ -1,7 +1,466 @@
+    # Compute the group commitment
+    group_commitment = compute_group_commitment(commitment_list, binding_factor_list)
+
+    # Compute the challenge
+    randomized_group_public_key = group_public_key + G * randomizer
+    challenge = compute_challenge(group_commitment, randomized_group_public_key, msg)
+
+    # Compute aggregated signature
+    z = Scalar(0)
+    for z_i in sig_shares:
+      z = z + z_i
+    return (group_commitment, z + randomizer * challenge), randomized_group_public_key
+
+
+The `verify_signature_share` is changed to incorporate the randomizer point,
+as follows: ::
+
+  Inputs:
+  - identifier, identifier i of the participant, a NonZeroScalar.
+  - PK_i, the public key for the i-th participant, where PK_i = G.ScalarBaseMult(sk_i),
+    an Element.
+  - comm_i, pair of Element values in G (hiding_nonce_commitment, binding_nonce_commitment)
+    generated in round one from the i-th participant.
+  - sig_share_i, a Scalar value indicating the signature share as produced in
+    round two from the i-th participant.
+  - commitment_list =
+      [(j, hiding_nonce_commitment_j, binding_nonce_commitment_j), ...], a
+    list of commitments issued in Round 1 by each participant, where each element
+    in the list indicates a NonZeroScalar identifier j and two commitment
+    Element values (hiding_nonce_commitment_j, binding_nonce_commitment_j).
+    This list MUST be sorted in ascending order by identifier.
+  - group_public_key, public key corresponding to the group signing key,
+    an Element.
+  - msg, the message to be signed, a byte string.
+  - randomizer_point, an element in G.
+
+  Outputs:
+  - True if the signature share is valid, and False otherwise.
+
+  def verify_signature_share(identifier, PK_i, comm_i, sig_share_i, commitment_list,
+                             group_public_key, msg, randomizer_point):
+    # Compute the randomized group public key
+    randomized_group_public_key = group_public_key + randomizer_point
+
+    # Compute the binding factors
+    binding_factor_list = compute_binding_factors(commitment_list, msg, randomizer_point)
+    binding_factor = binding_factor_for_participant(binding_factor_list, identifier)
+
+    # Compute the group commitment
+    group_commitment = compute_group_commitment(commitment_list, binding_factor_list)
+
+    # Compute the commitment share
+    (hiding_nonce_commitment, binding_nonce_commitment) = comm_i
+    comm_share = hiding_nonce_commitment + G.ScalarMult(binding_nonce_commitment, binding_factor)
+
+    # Compute the challenge
+    challenge = compute_challenge(group_commitment, randomized_group_public_key, msg)
+
+    # Compute the interpolating value
+    participant_list = participants_from_commitment_list(commitment_list)
+    lambda_i = derive_interpolating_value(identifier, participant_list)
+
+    # Compute relation values
+    l = G.ScalarBaseMult(sig_share_i)
+    r = comm_share + G.ScalarMult(PK_i, challenge * lambda_i)
+
+    return l == r
+
+
```

Comment:
Add the rationale / security analysis section here. (If there is separate rationale for the ciphersuites, add it in those sections.)